openshift-templates/hoverfly/template.yaml

```diff
@@ -0,0 +1,132 @@
+apiVersion: v1
+kind: Template
+labels:
+  template: hoverfly-template
+message: A Hoverfly service has been created in your project.
+metadata:
+  annotations:
+    description: |-
+      Hoverfly service
+    iconClass: icon-openshift
+    description: Hoverfly. A lightweight, opensource API simulation tool.
+    tags: instant-app
+    template.openshift.io/provider-display-name: SpectoLabs
+    template.openshift.io/support-url: https://hoverfly.io
+  name: hoverfly
+objects:
+- apiversion: v1
+  kind: Route
+  metadata:
+    name: ${HOVERFLY_SERVICE_NAME}
+    labels:
+      app: ${HOVERFLY_SERVICE_NAME}
+  spec:
+    to:
+      kind: Service
+      name: ${HOVERFLY_SERVICE_NAME}
+    port:
+      targetPort: 8500-http
+- apiVersion: v1
+  kind: Service
+  metadata:
+    name: ${HOVERFLY_SERVICE_NAME}
+    labels:
+      app: ${HOVERFLY_SERVICE_NAME}
+  spec:
+    ports:
+    - name: 8500-http
+      protocol: TCP
+      port: 8500
+      targetPort: 8500
+    - name: 8888-http
+      protocol: TCP
+      port: 8888
+      targetPort: 8888
+    selector:
+      app: ${HOVERFLY_SERVICE_NAME}
+      deploymentConfig: ${HOVERFLY_SERVICE_NAME}
+    type: ClusterIP
+    sessionAffinity: None
+- apiVersion: v1
+  kind: DeploymentConfig
+  metadata:
+    name: ${HOVERFLY_SERVICE_NAME}
+    labels:
+      app: ${HOVERFLY_SERVICE_NAME}
+  spec:
+    replicas: 1
+    selector:
+      app: ${HOVERFLY_SERVICE_NAME}
+      deploymentConfig: ${HOVERFLY_SERVICE_NAME}
+    strategy:
+      type: Recreate
+    template:
+      metadata:
+        labels:
+          app: ${HOVERFLY_SERVICE_NAME}
+          deploymentConfig: ${HOVERFLY_SERVICE_NAME}
+      spec:
+        containers:
+        - name: ${HOVERFLY_SERVICE_NAME}
+          image: ' '
+          ports:
+            - containerPort: 8500
+              protocol: TCP
+            - containerPort: 8888
+              protocol: TCP
+          livenessProbe:
+            httpGet:
+              path: /
+              port: 8500
+              scheme: HTTP
+            initialDelaySeconds: 120
+            timeoutSeconds: 1
+            periodSeconds: 10
+            sucessThreshold: 1
+            failureThreshold: 3
+          readinessProbe:
+            httpGet:
+              path: /
+              port: 8500
```

Comment:
please change to 8888 as hoverfly comes up in proxy mode and will not accept the non proxy requests to 8500. 8888 is the admin interface and will accept those requests.